timelock-server/src/main/java/com/palantir/atlasdb/timelock/paxos/PaxosTimeLockServer.java

```diff
@@ -121,6 +123,16 @@ private void registerLeaderElectionService(TimeLockServerConfiguration configura
                 localPaxosServices.ourLearner()));
         environment.jersey().register(new NotCurrentLeaderExceptionMapper());
         environment.jersey().register(new BlockingTimeoutExceptionMapper());
+
+        ServletRegistration.registerServlet(
+                environment,
+                ServletFactory.leaderPingServlet(localPaxosServices.pingableLeader()),
+                "/leader/ping");
+        ServletRegistration.registerServlet(
```

Comment:
One thing we could do here would be to have a (potentially static) class similar to `ServletRegistration` which handles the registration of all timelock resources. This thing could (a) enforce that the proper servlets are registered alongside the jersey registrations and (b) enforce that the proper object are registered (specifically thinking about checking that appropriate objects are wrapped in `AwaitingLeadershipProxy`).